03b_model/src/run_model.py

```diff
@@ -291,34 +581,183 @@ def plot_save_predictions(trainval_df, out_dir, run_id):
     
     #save predictions
     trainval_df.to_csv(os.path.join(out_dir, run_id, 'ModelResults.csv'))
+    
+def run_replicates(n_reps, prepped_model_io_data_file):
+    '''
+    run model replicates to understand variability
+    Parameters
+    ----------
+    n_reps : int
+        number of replicates to run
+    prepped_model_io_data_file : str
+        location of prepped model input, saved as a pickle file by write model params
 
+    Returns
+    -------
+    None.
 
-def main():
-    with open("03b_model/model_config.yaml", 'r') as stream:
-        config = yaml.safe_load(stream)
+    '''
     
-    inputs = config['inputs']
-    target = config['target']
-    
-    out_dir = config['out_dir']
-    run_id = config['run_id']
+    for i in range(config['replicates']):
+        
+        run_id = os.path.join(config['run_id'], str(i).rjust(2,'0'))
+        
+        train_model(prepped_model_io_data_file, inputs, seq_len,
+                        hidden_units, recur_dropout, 
+                        dropout, n_epochs, learn_rate, 
+                        out_dir, run_id,                       
+                        train_start_date, train_end_date,
+                        val_start_date, val_end_date,
+                        test_start_date, test_end_date, inc_ante)
+        
+        predictions = make_predictions(prepped_model_io_data_file, 
+                             hidden_units, recur_dropout, dropout, 
+                             n_epochs, learn_rate, out_dir, run_id,
+                             train_start_date, train_end_date,
+                             val_start_date, val_end_date,
+                             test_start_date, test_end_date)
+        
+        plot_save_predictions(predictions, out_dir, run_id)
+        
+
+def test_hyperparameters():
```

Comment:
Noting missing function description. Would help to note what kind of tuning is done (looks like grid search based on input values?)

Is this function currently unused?

Reply:
I updated the docstring in the latest commit. The function is used but not within the `main()` call, I call it directly to run the hyperparameter search